### ISSUE TITLE
🔎 SEO - Pages not being indexed - Part 1

### ISSUE THREAD

**Issue by Calinator444** [SSWConsulting/SSW.Website#2811]
From: @wicksipedia
Sent: Monday, June 24, 2024 11:20 AM
To: @andrewwaltosssw
Cc: @sethdaily ; SSW Website v3 [SSWWebsiteV3@ssw.com.au](mailto:SSWWebsiteV3@ssw.com.au); @camillars; Jeoffrey Fischer [SSW] [JeoffreyFischer@ssw.com.au](mailto:JeoffreyFischer@ssw.com.au)
Subject: Re: New reasons prevent pages in a sitemap from being indexed on site https://www.ssw.com.au/

### Description
Search Console has identified that some pages on our website are not being indexed.

URL
https://search.google.com/u/1/search-console/index?resource_id=https://www.ssw.com.au/&utm_source=wnc_20237597&utm_medium=gamma&utm_campaign=wnc_20237597&utm_content=msg_110624660&hl=en
### Solution
Fix the following issues.

- [x] Alternate page with proper canonical tag - 26,094 pages
- [x] Page with redirect - 6,543 pages
- [x] Not found (404) - 2,778 pages
- [x] Excluded by ‘noindex’ tag - 1,666 pages
- [x] Duplicate without user-selected canonical - 1,051 pages

### Acceptance Criteria

- [x] All issues are fixed

**Comment by Calinator444** [comment]
> Alternate page with proper canonical tag - 26,094 pages

All of the new pages that were listed as **Alt page with proper canonical tag** were browse pages.

Essentially these browse pages list the pages at the base of each route. They don't have any unique content and they'll be removed when the VM gets switched off. I'm not sure why they're being de-indexed but they'll be removed soon anyway so it's not worth trying to re-index them.


![Image](https://github.com/SSWConsulting/SSW.Website/assets/65635198/1936db06-a610-4ac2-8404-62d5df482908)

Google also occasionally selects a different variant of a url
e.g. 
https://www.ssw.com.au/people/zach-keeping/ (🔍 google selected canonical URL https://ssw.com.au/people/zach-keeping/)


> Page with redirect - 6,543 pages

Nothing to action here, I didn't notice anything out the ordinary. We have a lot of redirects set up on the site to link people to our consulting pages if they try to click on an older URL. Aside from that most of the pages being omitted because google found the proper canonical URLs (e.g. "https://www.ssw.com.au/rules//email-send-a-v2" points to "https://www.ssw.com.au/rules/email-send-a-v2/".

> Not Found (404) https://github.com/SSWConsulting/SSW.Rules/issues/1391

A large number of the rules pages contain broken links. This appears to be an Issue with Gatsby. I've logged this here https://github.com/SSWConsulting/SSW.Rules/issues/1391.


I also found that a number of the archive pages had links that would 404 because they were referencing pages on the v1 site using relative paths. Adding redirects for these in Cloudflare won't work because the path to the file in `/archive` doesn't match the original path. (e.g. href = "UserGuide.aspx" would point to `ssw.com.au/archive/UserGuide.aspx` instead of `ssw.com.au/ssw/UserGuide.aspx`
Here's a fix for this (note: we'll still need the cloudflare redirects in place to fix these 404s): https://github.com/SSWConsulting/SSW.Website-v1-Progress/pull/142)

> Excluded by ‘noindex’ tag - 1,666 pages

Most of these are on the v1 site, however I've created a [PR ](https://github.com/SSWConsulting/SSW.Website-v1-Progress/pull/141)for stripping the no index tags on the archived pages that had them.

Some of the rules pages were also not being indexed but this is a known issue that has beene fixed: https://github.com/SSWConsulting/SSW.Rules/issues/1285

> Duplicate without User selected canonical

There's a few causes for this:

- All of the old newsletters from the site were ported over to the Next js application, which serves them statically. However these pages were not deleted from the v1 server. 

e.g. 
selected canonical url: https://www.ssw.com.au/images/newsletter-uploads/2008/_2008_10_NZ_Dunedin%20User%20Group_.htm

non-indexed url: https://www.ssw.com.au/ssw/netug/SSWUpdate/_2008_10_NZ_Dunedin User Group_.htm

- The ASP.Net server will serve urls that are both lower case and upper case. Google often prefers the lower case URL and indexes it over the non-lower case version.

- Google sometimes tries to map an existing URL with a query parameter. However because a page with a parameter is the same as the page without the parameter google sees a duplicate.

**Comment by Calinator444** [comment]
TLDR: 

> Alternate page with proper canonical tag - 26,094 pages

Not done - Google has been favoring certain variants of urls that link to the same page. Others don't get indexed. We may need to fix this using cloudflare.

> Not Found (404) 

Done- Logged as an [issue ](https://github.com/SSWConsulting/SSW.Rules/issues/1391)with Gatsby. Also [fixed ](https://github.com/SSWConsulting/SSW.Website-v1-Progress/pull/142)some of the links in the archive pages.

> Excluded by 'noindex' tag

Done - Stripped some noindex tags from the archive pages ([PR](https://github.com/SSWConsulting/SSW.Website-v1-Progress/pull/141). There's also a known issue for rules causing this https://github.com/SSWConsulting/SSW.Rules/issues/1285.

> Page with redirect - 6,543 pages

Not Done - These redirects appear to be coming from front door, or our server just fixes a URL and marks it as a redirect (www.ssw.com.au => www.ssw.com.au/)


> Duplicate without User selected canonical

Not Done:
- The newsletters have been migrated but the old links are still active
- Google tried to map some existing pages with query params


I've added a list of known isues in SharePoint
https://sswcom.sharepoint.com/:w:/r/sites/SSWMarketing/_layouts/15/Doc.aspx?sourcedoc=%7B8F4C361D-1664-45FE-B10C-A7EE3E9A30A4%7D&file=Google%20Analytics%20-%20known%20issues%20for%20indexing%20pages.docx&action=default&mobileredirect=true&wdsle=0
